### DISCUSSION TITLE
Cannot get url in getServerSideProps

### DISCUSSION THREAD

**Discussion by madebyais** [vercel/next.js#43616] [Help] [answered]
Hi team,

I need your help since I was unable to get the url in the NextPageContext when using getServerSideProps.

Here's my custom code:
```
export const ConsolePageContext = async (ctx: NextPageContext, fn?: Function) => {
  ....
  let { profile }: any = token
  let props: any = { profile, query: ctx.query }

  if (fn) {
    const fnResult: any = await fn(ctx, props)
    if (fnResult.redirect) return { ...fnResult }
    props = { ...props, ...fnResult.props }
  }

  console.dir(ctx.req?.url)

  return { props }
}
```

Result:
```
...
[Symbol(NextRequestMeta)]: {
    __NEXT_INIT_URL: 'http://localhost:3000/_next/data/development/teams.json',
    __NEXT_INIT_QUERY: {},
    _protocol: 'http',
    __NEXT_CLONABLE_BODY: {
      finalize: [AsyncFunction: finalize],
      cloneBodyStream: [Function: cloneBodyStream]
    },
    __nextHadTrailingSlash: false,
    __nextIsLocaleDomain: false
  }
  ```

The `ctx.req?.url` returns `'/_next/data/development/teams.json'` instead of `/teams`. 
However, it works when I refresh the page.

For your information I'm using this approach in the route `pages/teams/[[...paths]].tsx`

Any advice on this? Thanks!

**Comment by icyJoseph** [comment, marked as answer]
Try with: 

> resolvedUrl: A normalized version of the request URL that strips the _next/data prefix for client transitions and includes original query values.

Taken from https://nextjs.org/docs/api-reference/data-fetching/get-server-side-props#context-parameter.

**Reply by madebyais** [reply]
Awesome, your suggestion solved the issue. Thanks a lot @icyJoseph !
